lib/flowViewport/API/interfacesImp/fvpInformationInterfaceImp.cpp

```diff
@@ -0,0 +1,99 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//Local headers
+#include "fvpInformationInterfaceImp.h"
+#include "flowViewport/API/perViewportSceneIndicesData/fvpViewportInformationAndSceneIndicesPerViewportDataManager.h"
+
+#include <mutex>
+
+namespace{
+    std::mutex _viewportInformationClient_mutex;
+}
+    
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace FVP_NS_DEF {
+
+//Set of information clients
+static InformationClientSet _viewportInformationClients;
+
+InformationInterface& InformationInterface::Get() 
+{ 
+    return InformationInterfaceImp::Get();
+}
+
+InformationInterfaceImp& InformationInterfaceImp::Get()
+{
+    static InformationInterfaceImp theInterface;
+    return theInterface;
+}
+
+void InformationInterfaceImp::RegisterInformationClient(const InformationClient& client)
+{
+    std::lock_guard<std::mutex> lock(_viewportInformationClient_mutex);
+
+    InformationClient* clientNonConst = const_cast<InformationClient*>(&client);
+    auto foundResult = _viewportInformationClients.find(clientNonConst);
+    if (foundResult == _viewportInformationClients.cend()){
+        _viewportInformationClients.insert(clientNonConst);
+    }
+}
+
+void InformationInterfaceImp::UnregisterInformationClient(const InformationClient& client)
+{
+    std::lock_guard<std::mutex> lock(_viewportInformationClient_mutex);
+
+    InformationClient* clientNonConst = const_cast<InformationClient*>(&client);
+    auto foundResult = _viewportInformationClients.find(clientNonConst);
+    if (foundResult != _viewportInformationClients.end()){
+        _viewportInformationClients.erase(foundResult);
+    }
+}
+
+void InformationInterfaceImp::SceneIndexAdded(const InformationInterface::ViewportInformation& _viewportInfo)
+{
+    std::lock_guard<std::mutex> lock(_viewportInformationClient_mutex);
+    for (auto viewportInfoClient : _viewportInformationClients){
+        if (viewportInfoClient){
+            viewportInfoClient->SceneIndexAdded(_viewportInfo);
+        }
+    }
+}
+
+void InformationInterfaceImp::SceneIndexRemoved(const InformationInterface::ViewportInformation& _viewportInfo)
+{
+    std::lock_guard<std::mutex> lock(_viewportInformationClient_mutex);
+    for (auto viewportInfoClient : _viewportInformationClients){
+        if (viewportInfoClient){
+            viewportInfoClient->SceneIndexRemoved(_viewportInfo);
+        }
+    }
+}
+
+void InformationInterfaceImp::GetViewportsInformation(std::set<const ViewportInformation*>& outHydraviewportInformationSet)const
+{
+    outHydraviewportInformationSet.clear();
+    const ViewportInformationAndSceneIndicesPerViewportDataSet& viewportInformationAndSceneIndicesPerViewportDataVector = 
```

Comment:
This variable still has vector in its name but is now a set

Reply:
I think this is where we should be careful with long identifier names: they should not reveal implementation details, which is violating encapsulation and Don't Repeat Yourself.

Reply:
I think using Array in the name can still be confusing since arrays are also a specific container. What I tend to go with is using plurals, but since this variable/class name is already quite long, a lone "s" will be hard to spot and differentiate. Maybe we could name it something like allViewportInformationAndSceneIndicesPerViewportData? Ultimately I don't have that strong of an opinion on this.

Reply:
Right, I changed it  in allViewportInformationAndSceneIndicesPerViewportData.